commit de7828ae46f6feb9a219e9b1444e1d7e32bd94ac
Author: Volker Krause <vkrause@kde.org>
Date:   Thu Aug 3 19:47:59 2017 +0200

    Add cross-compilation support for the highlighting indexer
    
    Reviewers: #frameworks
    
    Tags: #frameworks
    
    Differential Revision: https://phabricator.kde.org/D7102

diff --git a/src/indexer/CMakeLists.txt b/src/indexer/CMakeLists.txt
index a2c7ef7..acb1647 100644
--- a/src/indexer/CMakeLists.txt
+++ b/src/indexer/CMakeLists.txt
@@ -1,7 +1,13 @@
-add_executable(katehighlightingindexer katehighlightingindexer.cpp)
-
-if(Qt5XmlPatterns_FOUND)
-  target_link_libraries(katehighlightingindexer Qt5::XmlPatterns)
+if(CMAKE_CROSSCOMPILING AND KATEHIGHLIGHTINGINDEXER_EXECUTABLE)
+    # cross build, we need the indexer supplied externally
+    add_executable(katehighlightingindexer IMPORTED GLOBAL)
+    set_target_properties(katehighlightingindexer PROPERTIES IMPORTED_LOCATION ${KATEHIGHLIGHTINGINDEXER_EXECUTABLE})
 else()
-  target_link_libraries(katehighlightingindexer Qt5::Core)
+    # host build
+    add_executable(katehighlightingindexer katehighlightingindexer.cpp)
+    if(Qt5XmlPatterns_FOUND)
+        target_link_libraries(katehighlightingindexer Qt5::XmlPatterns)
+    else()
+        target_link_libraries(katehighlightingindexer Qt5::Core)
+    endif()
 endif()
